### ISSUE TITLE
replace calls to system and shell in r4ss with system2

### ISSUE THREAD

**Issue by k-doering-NOAA** [r4ss/r4ss#456]
Throughout the r4ss package where calls to Stock Synthesis are made, `system` or `shell` are used depending on the operating system or the user's request. (For example, See the [SS_doRetro code](https://github.com/r4ss/r4ss/blob/development/R/SS_doRetro.R#L167-L172)). I just learned about the [system2 function](https://www.rdocumentation.org/packages/base/versions/3.6.2/topics/system2), which seems like a simpler replacement for system and shell and would work on any OS. Perhaps we should overhaul the r4ss code and just used the system2 function? I think it would simplify the code base for functions requiring the executable.

**Comment by iantaylor-NOAA** [comment]
This is a good idea. It looks like `system()` and `shell()` occur in the following files
- NegLogInt_Fn.R
- run_SS_models.R
- SSbootstrap.R
- SS_doRetro.R
- SS_profile.R
- SS_RunJitter.R
and there's inconsistent use of `CallType` or `systemcmd` arguments to override the defaults. It would seem that `system2()` would clean up the code and allow those arguments to be removed.

**Comment by iantaylor-NOAA** [comment]
I started this in commit 50a544d to new branch `system2_issue456` but I failed to note that the `intern = TRUE/FALSE` argument used with `system()` has been replaced by the argument `stdout` in `system2()`. Rather than make similar changes across a bunch of files to adapt to that new approach, I think this issue should be combined with issue #430 and rely on `run_SS_models()` to take care of the following steps:

- sort out Windows vs Unix use of "./"
- deal with exe in path vs local directory
- align inconsistent arguments that are `exe`, `exefile`, or `model`, depending on the function
- control whether command line info from model run is saved to a file (as requested by `intern = TRUE` or print to the R console)

**Comment by k-doering-NOAA** [comment]
Good idea to work on issue #430 at the same time! I think all the calls to system2 would be just to run SS, so it makes sense to use `run_SS_models` to reduce redundancy

**Comment by kellijohnson-NOAA** [comment]
What do you think the timeline of switching is? I have a proposed change for the call in `run_SS_models` to decrease the output to the screen [from](https://github.com/r4ss/r4ss/blob/2e5f0d5349c4e269fc6018be81f95f137360ff6c/R/run_SS_models.R#L141)
```
console.output <- system(command, intern = intern)
```
to
```
console.output <- system(command, intern = intern, show.output.on.console = verbose)
```
but I am not sure if it is worth it if we are changing. It is just hard in ss3sim to see what is going wrong in tests when you have to scroll through all of the output that is printed to the screen.

**Comment by iantaylor-NOAA** [comment]
@kellijohnson-NOAA go ahead and make the change to get the extra output out of the way in the short term.

I spent a while in late December pursuing the switch from `system()` & `shell()` to `system2()` (shown in commit 50a544de9caaef269e53b11814f9a4f2b048e372 to the `system2_issue456` branch) before realizing that a more comprehensive solution as described in #460 would be a better approach. However, that larger revision was taking more time than I had available in early January so I never finished it. I just pushed commit 300cbe61938c0843ceb854f3ae9a37565aab1ccc which has my work in progress for the `run_SS_models()` function in case you have time to review my changes. I don't even know if the function works in that state but at least you can see the changes in arguments I was making. Once that function is sorted, and also renamed to something simpler like `runSS()`, all the functions that call SS can be cleaned up to just call this one.

**Comment by kellijohnson-NOAA** [comment]
Great, I pushed the change to killoutput c2d76d6 and I will submit a pull request once it passes.

**Comment by kellijohnson-NOAA** [comment]
Thanks to @k-doering-NOAA for providing the link to this [informative blog post](https://ropensci.org/blog/2021/09/13/system-calls-r-package/) about `system()`, `system2()`, `sys::exec_internal()`, and others. The main benefit of other functions appears to be when running things in parallel you can advance R without needing to wait for the system call to finish.

**Comment by iantaylor-NOAA** [comment]
Implemented in Pull Request #722.
